expression/builtin_encryption.go

```diff
@@ -494,12 +494,17 @@ type sha2FunctionClass struct {
 }
 
 func (c *sha2FunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinSHA2Sig{newBaseBuiltinFunc(args, ctx)}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString, tpInt)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	types.SetBinChsClnFlag(bf.tp)
```

Comment:
1. You should do a type assertion, which is `const, ok := bfArgs[1].(*expression.Constant)`, before call `bfArgs[1].EvalInt(...)`.
2. If assertion succeed but the constant value is not in set {0, 224, 256 ...}, just set `Flen` to 0. **Do not** return a nil Expression in this situation
3. If assertion failed, which indicate that the parameter is not a constant value but a column reference instead, you can just set `Flen` to 512/4